text/0156-columnar-encoding.md

```diff
@@ -0,0 +1,1031 @@
+- A set of data points exemplars (also adhering to a one-to-many relationship).
+
+Exemplar entities, in turn, are connected to their dedicated set of attributes.
+
+Technically speaking, the `quantile` entity isn't encoded as an independent entity but rather as a list of struct within
+the `SUMMARY_DATA_POINTS entity`.
+
+![Metrics Arrow Schema](img/0156_metrics_schema.png)
+
+Gauge and Sum are identified by the `metric_type` column in the `METRICS` entity and they share the same Arrow record
+for the data points, i.e. `NUMBER_DATA_POINTS`.
+
+`span_id` and `trace_id` are represented as fixed size binary dictionaries by default but can evolve to non-dictionary
+form when their cardinality exceeds a certain threshold (usually 2^16).
+
+As usual, each of these Arrow records is sorted by specific columns to optimize the compression ratio. With this mapping
+batch of metrics containing a large number of data points sharing the same attributes and timestamp will be highly
+compressible (multivariate time-series scenario).
+
+> Note: every OTLP timestamps are represented as Arrow timestamps with nanoseconds time unit. This representation will
+> simplify the integration with the rest of the Arrow ecosystem (numerous time/date functions are supported in
+> DataFusion for example).
+> Note: aggregation_temporality is represented as an Arrow dictionary with a dictionary index of type int8. This OTLP
+> enum has currently 3 variants, and we don't expect to have in the future more than 2^8 variants.
+
+## Implementation Recommendations
+
+### Protocol Extension and Fallback Mechanism
+
+The support of this new protocol can only be progressive, so implementers are advised to follow the following
+implementation recommendations in phase 1:
+
+* OTLP Receiver: Listen on a single TCP port for both OTLP and OTel Arrow. The goal is to make the support of this
+  protocol extension
+  transparent and automatic. This can be achieved by adding the `ArrowStreamService` to the same gRPC listener. A
+  configuration
+  parameter will be added to the OTLP receiver to disable this default behavior to support specific uses.
+* OTLP Exporter: By default the OTLP exporter should initiate a connection to the `ArrowStreamService` endpoint of the target
+  receiver. If this connection fails because the `ArrowStreamService` is not implemented by the target, the exporter
+  must automatically fall back on the behavior of the classic OTLP protocol. A configuration parameter could be added to
+  disable this default behavior.
+
+The implementation of these two rules should allow a seamless and
+adaptive integration of OTel Arrow into the current ecosystem
+generally.
+
+For the prototype specifically, which is a fork of the OpenTelemetry
+collector codebase, we have derived the OTLP/gRPC-Arrow exporter and
+receiver as set of changes directly to the `receiver/otlpreceiver` and
+`exporter/otlpexporter` components, with new `internal/arrow` packages
```

Comment:
To avoid confusion I think on the prototype repo as well it would be useful to use `otelarrow` as the name of receiver and exporter.

Reply:
@jmacd that sounds reasonable to me.

Reply:
Done